src/Functions/FunctionsConversion.h

```diff
@@ -2132,7 +2132,11 @@ inline bool isDateTime64(const ColumnsWithTypeAndName & arguments)
     return false;
 }
 
-template <typename ToDataType, typename Name, typename MonotonicityImpl>
+template <
+    typename ToDataType,
+    typename Name,
+    typename MonotonicityImpl,
+    FormatSettings::DateTimeOverflowBehavior template_date_time_overflow_behavior = default_date_time_overflow_behavior>
```

Comment:
This doesn't seem to benefit at all from being a template parameter. Have you considered a constructor argument instead?